db/osqlcomm.c

```diff
@@ -5121,106 +5115,86 @@ int osql_comm_signal_sqlthr_rc(osql_sess_t *sorese, struct errstat *xerr,
         return 0;
 
     /* if error, lets send the error string */
-    if (sorese->host) {
-        /* remote */
-        if (sorese->rqid == OSQL_RQID_USE_UUID) {
-            osql_done_xerr_uuid_t rpl_xerr = {{0}};
-            osql_done_uuid_rpl_t rpl_ok = {{0}};
-
-            if (rc) {
-                uint8_t *p_buf = buf;
-                uint8_t *p_buf_end = buf + OSQLCOMM_DONE_XERR_UUID_RPL_LEN;
-                rpl_xerr.hd.type = OSQL_XERR;
-                comdb2uuidcpy(rpl_xerr.hd.uuid, sorese->uuid);
-                rpl_xerr.dt = *xerr;
-
-                osqlcomm_done_xerr_uuid_type_put(&(rpl_xerr), p_buf, p_buf_end);
-                logmsg(LOGMSG_DEBUG,
-                       "%s line %d master signaling %s uuid %s with rc=%d "
-                       "xerr=%d\n",
-                       __func__, __LINE__, sorese->host,
-                       comdb2uuidstr(sorese->uuid, uuid), rc, xerr->errval);
-
-                msglen = OSQLCOMM_DONE_XERR_UUID_RPL_LEN;
-
-            } else {
-                uint8_t *p_buf = buf;
-                uint8_t *p_buf_end = buf + OSQLCOMM_DONE_UUID_RPL_LEN;
+    if (!sorese->host) {
+        /* local */
+        return osql_chkboard_sqlsession_rc(sorese->rqid, sorese->uuid,
+                                           sorese->nops, NULL, xerr);
+    }
 
-                rpl_ok.hd.type = OSQL_DONE;
-                comdb2uuidcpy(rpl_ok.hd.uuid, sorese->uuid);
-                rpl_ok.dt.rc = 0;
-                rpl_ok.dt.nops = sorese->nops;
+    /* remote */
+    if (sorese->rqid == OSQL_RQID_USE_UUID) {
+        osql_done_xerr_uuid_t rpl_xerr = {{0}};
+        osql_done_uuid_rpl_t rpl_ok = {{0}};
 
-                osqlcomm_done_uuid_rpl_put(&(rpl_ok), p_buf, p_buf_end);
+        if (rc) {
+            msglen = OSQLCOMM_DONE_XERR_UUID_RPL_LEN;
+            uint8_t *p_buf = buf;
+            uint8_t *p_buf_end = buf + msglen;
+            rpl_xerr.hd.type = OSQL_XERR;
+            comdb2uuidcpy(rpl_xerr.hd.uuid, sorese->uuid);
+            rpl_xerr.dt = *xerr;
 
-                logmsg(LOGMSG_DEBUG,
-                       "%s line %d master signaling %s uuid %s with rc=%d "
-                       "xerr=%d\n",
-                       __func__, __LINE__, sorese->host,
-                       comdb2uuidstr(sorese->uuid, uuid), rc, xerr->errval);
+            osqlcomm_done_xerr_uuid_type_put(&(rpl_xerr), p_buf, p_buf_end);
 
-                msglen = OSQLCOMM_DONE_RPL_LEN;
```

Comment:
This was a bug probably, should have been OSQLCOMM_DONE_UUID_RPL_LEN.

Reply:
It most certainly was. I tried to fix it as well (https://github.com/bloomberg/comdb2/pull/1868).